### ISSUE TITLE
Fix version/attribute inconsistencies in output VCFs.

### ISSUE THREAD

**Issue by cmnbroad** [broadinstitute/gatk#1116]
Its possible to produce a version 4.2 VCF with a GQ FORMAT field that specifies INTEGER but with floating point GQ values if the variants are read from an older version file and are not fully decoded.(GATK3 ticket is (https://github.com/broadinstitute/gsa-unstable/issues/1060). Rather than forward porting the GATK fix in https://github.com/broadinstitute/gsa-unstable/pull/1204 we should find a more general solution for HB.

**Comment by droazen** [comment]
@cmnbroad Is this one done?

**Comment by droazen** [comment]
Can we address this at the htsjdk level?

**Comment by droazen** [comment]
@cmnbroad Is this one still relevant?

**Comment by cmnbroad** [comment]
@droazon Yes, I think we should keep this open. The VCF header refactoring will be step one on the road to fixing this.
